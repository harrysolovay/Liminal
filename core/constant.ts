import { Ty } from "./Ty.ts"

export type constant<V extends number | string> = ReturnType<typeof constant<V>>

export function constant<V extends number | string>(value: V): Ty<V, never> {
  return Ty((description) => ({
    const: value,
    description,
  }))
}
