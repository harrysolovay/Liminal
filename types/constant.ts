import { Ty } from "./Ty.ts"

export function constant<V extends number | string>(value: V): Ty<V, never> {
  return Ty((description) => ({
    const: value,
    description,
  }))
}
