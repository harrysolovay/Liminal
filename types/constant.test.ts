import { constant } from "./constant.ts"
import { assertTySnapshot } from "test_util"

Deno.test("constant num", async (t) => {
  await assertTySnapshot(t, constant("Hello"))
})

Deno.test("constant str", async (t) => {
  await assertTySnapshot(t, constant(42))
})
