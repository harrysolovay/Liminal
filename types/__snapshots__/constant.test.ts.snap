export const snapshot = {};

snapshot[`constant num 1`] = `
{
  additionalProperties: false,
  description: "",
  properties: {
    value: {
      const: "Hello",
      description: "",
    },
  },
  required: [
    "value",
  ],
  type: "object",
}
`;

snapshot[`constant num 2`] = `
{
  additionalProperties: false,
  description: "",
  properties: {
    value: {
      const: "Hello",
      description: "Description one.",
    },
  },
  required: [
    "value",
  ],
  type: "object",
}
`;

snapshot[`constant num 3`] = `
{
  additionalProperties: false,
  description: "",
  properties: {
    value: {
      const: "Hello",
      description: "Description two. Description one.",
    },
  },
  required: [
    "value",
  ],
  type: "object",
}
`;

snapshot[`constant str 1`] = `
{
  additionalProperties: false,
  description: "",
  properties: {
    value: {
      const: 42,
      description: "",
    },
  },
  required: [
    "value",
  ],
  type: "object",
}
`;

snapshot[`constant str 2`] = `
{
  additionalProperties: false,
  description: "",
  properties: {
    value: {
      const: 42,
      description: "Description one.",
    },
  },
  required: [
    "value",
  ],
  type: "object",
}
`;

snapshot[`constant str 3`] = `
{
  additionalProperties: false,
  description: "",
  properties: {
    value: {
      const: 42,
      description: "Description two. Description one.",
    },
  },
  required: [
    "value",
  ],
  type: "object",
}
`;
